aflr/api_request.py

```diff
@@ -7,49 +7,61 @@
 
 
 class APIRequest:
-    def __init__(self, api_key=None, api_base=None, **params):
-        self.api_key = api_key if api_key else aflr.api_key
-        self.api_base = api_base if api_base else aflr.api_base
-        if self.api_key == None or self.api_key == "your-key":
-            self.api_key = os.environ.get("aflr_key", None)
-        if self.api_key == None or len(self.api_key) < 30:
+    def _api_key_checker(api_key=None):
+        if api_key == None or api_key == "your-key":
+            api_key = os.environ.get("aflr_key", None)
+        if not isinstance(api_key, str):
+            raise TypeError("api_key must be of type string.")
+        if len(api_key) < 32:
             raise ValueError(
-                "Please specify a valid api_key or create one here:\nhttps://console.api.audio"
+                "Please specify a valid api_key or create one here:\n https://console.api.audio"
             )
-        if not isinstance(self.api_key, str):
-            raise TypeError("api_key must be of type string.")
 
-    def _build_header(self):
+    @classmethod
+    def _build_header(cls):
+        cls._api_key_checker(aflr.api_key)
         # add more headers for analytics in the future
-        return {"x-api-key": self.api_key}
+        return {"x-api-key": aflr.api_key}
 
-    def _post_request(self, json, url=None):
-        url = self.url if not url else url
-        headers = self._build_header()
+    @classmethod
+    def _post_request(cls, json, url=None):
+        url = url or f"{aflr.api_base}{cls.resource_path}"
+        headers = cls._build_header()
```

Comment:
lets have new lines here as well like in the _get_request

Reply:
sure, added a convention :)